### PR TITLE
add match algorithm option support for face matching

### PR DESCRIPTION
**Pull request by maozhong1**
Currently by default hungarian assignment(linear_sum_assignment) is used when do face match.   Some customers prefer using "mininum cosion distance" for face matching in some use cases.  This pull request created to add this option.

### CODE REVIEW

File: demos/python_demos/face_recognition_demo/face_recognition_demo.py [hunk 1]
```diff
@@ -55,6 +56,8 @@ def build_argparser():
                          help="(optional) Crop the input stream to this height " \
                          "(default: no crop). Both -cw and -ch parameters " \
                          "should be specified to use crop.")
+    general.add_argument('-match_algo', default='HUNGARIAN', choices=MATCH_ALGO,
```
Comment (IRDonch) [inline demos/python_demos/face_recognition_demo/face_recognition_demo.py]: Long options should start with two dashes.

Comment (maozhong1) [reply demos/python_demos/face_recognition_demo/face_recognition_demo.py]: i add '--match_algo' for long option, thanks.

**Comment by maozhong1** [comment]
--match_algo is added. more ,I modified code in faces_database.py"min_dist = distances[i][id]" to further optimization.

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Pull request merged**